CONTRIBUTING.md

```diff
@@ -1,31 +1,117 @@
+The XRP Ledger has many and diverse stakeholders, and everyone deserves
+a chance to contribute meaningful changes to the code that runs the XRPL.
+
 # Contributing
-The XRP Ledger has many and diverse stakeholders, and everyone deserves a chance to contribute meaningful changes to the code that runs the XRPL.
-To contribute, please:
-1. Fork the repository under your own user.
-2. Create a new branch on which to write your changes. Please note that changes which alter transaction processing must be composed via and guarded using [Amendments](https://xrpl.org/amendments.html). Changes which are _read only_ i.e. RPC, or changes which are only refactors and maintain the existing behaviour do not need to be made through an Amendment.
-3. Write and test your code.
-4. Ensure that your code compiles with the provided build engine and update the provided build engine as part of your PR where needed and where appropriate.
-5. Write test cases for your code and include those in `src/test` such that they are runnable from the command line using `./rippled -u`. (Some changes will not be able to be tested this way.)
-6. Ensure your code passes automated checks (e.g. clang-format and levelization.)
-7. Squash your commits (i.e. rebase) into as few commits as is reasonable to describe your changes at a high level (typically a single commit for a small change.)
-8. Open a PR to the main repository onto the _develop_ branch, and follow the provided template.
-
-# Major Changes
-If your code change is a major feature, a breaking change or in some other way makes a significant alteration to the way the XRPL will operate, then you must first write an XLS document (XRP Ledger Standard) describing your change.
-To do this:
-1. Go to [XLS Standards](https://github.com/XRPLF/XRPL-Standards/discussions).
-2. Choose the next available standard number.
-3. Open a discussion with the appropriate title to propose your draft standard.
-4. Link your XLS in your PR.
+
+We assume you are familiar with the general practice of [making contributions
+on GitHub][1].
+This file includes only special instructions specific to this project.
+
+
+## Before you start
+
+All of your contributions must be developed in your personal
+[fork](https://github.com/XRPLF/rippled/fork).
+No personal branches may ever be pushed to the [main project][rippled].
+These are the only branches that may ever exist in the main project:
+
+- `develop`: The latest set of unreleased features, and the most common
+    starting point for contributions.
+- `release`: The latest release candidate.
+- `master`: The latest stable release.
+- `gh-pages`: The documentation for this project, built by Doxygen.
+
+All of your commits must be signed,
```

Comment:
Might also be worth having this checked by a bot in addition to stating it here.

Reply:
I don't think GitHub's built-in protections can enforce this, and I'm not aware of a third-party bot that does it, but it is somewhat enforced by the fact that PRs cannot be squashed and merged unless all commits are signed.

Reply:
GitHub's protections can enforce that the tip of any branch is signed. I don't think it's necessary that _all_ commits are signed as long as the tip is. My understanding is that signing a commit is implicitly verifying all the commits leading up to it (sort of how validating a ledger in rippled implicitly validates all the previous ledgers). It may be worth rephrasing this to something like "All of your commits should be signed. The tip of pushed branches must be signed."

Reply:
I'll add this explanation to the document.